lms/services/vitalsource/_client.py

```diff
@@ -97,6 +179,43 @@ def _json_request(self, method, url):
             method, url, headers={"Accept": "application/json"}
         )
 
+    def _xml_request(self, method, url, data=None, **kwargs) -> dict:
+        """
+        Make a request to a VitalSource endpoint that accepts/returns XML.
+
+        The VitalSource API endpoints prefixed with "v3/" use XML.
+        """
+        kwargs["headers"] = {"Accept": "text/xml"}
+
+        if data:
+            kwargs["data"] = xmltodict.unparse(data)
+            kwargs["headers"]["Content-Type"] = "text/xml"
+
+        response = self._http_session.request(method, url, **kwargs)
+        return xmltodict.parse(response.text)
+
+
+class _VSUserAuth(AuthBase):
+    """
+    A requests authentication method for user based access tokens.
+
+    See: https://requests.readthedocs.io/en/latest/user/advanced/#custom-authentication
+
+    Using this allows us to completely separate authentication from other
+    behaviors which leads to simpler tests, and code. Authentication is also
+    transparent to the caller and results can be cached.
+    """
+
+    def __init__(self, client: VitalSourceClient, user_reference: str):
+        self._client = client
```

Comment:
I think the dependency of client here for `get_user_credentials` is not very satisfying at fist glance but I can't see a better alternative.

Reply:
Yeah I tried a few things. One option is to create the object within a closure in a function:

```python
def user_auth(self, user_reference):
    def _auth(request):
        request.headers['Blah'] = self.blah
    return _auth
```

As technically, any callable will do. This works, but is a bit weird, and makes testing really funky. Have a separate object:

* Allows us to use our normal patching, instead of doing `patch.object(client, _auth)`
* Allows us to follow the examples in the docs, for whatever that's worth